lib/msf/core/web_services/servlet/session_event_servlet.rb

```diff
@@ -15,17 +14,22 @@ def self.registered(app)
 
   #######
   private
+
   #######
 
   def self.get_session_event
     lambda {
       warden.authenticate!
       begin
         sanitized_params = sanitize_params(params, env['rack.request.query_hash'])
-        data = get_db.session_events(sanitized_params)
+        data, pagination_total = get_db.session_events(sanitized_params)
         data = data.first if is_single_object?(data, sanitized_params)
-        set_json_data_response(response: data)
-      rescue => e
+        if pagination_total.nil?
+          set_json_data_response(response: data)
+        else
+          set_json_data_response(response: data, merge_data: { 'pagination_total' => pagination_total })
```

Comment:
I think it would be better to have the merging of the pagination total done here rather than in the`set_json_data_response` function

Reply:
As an example, I think that would look like:

```suggestion
        else
          data = data.merge({ 'pagination_total' => pagination_total })
          set_json_data_response(response: data)
```

And reverting the `set_json_data_response` back to its previous implementation:

```diff
-  def set_json_data_response(response:, includes: nil, code: 200, merge_data: {})
+  def set_json_data_response(response:, includes: nil, code: 200)
```

Reply:
No, I said before that data cannot be merged with paging, data is an array of "Mdm::SessionEvent::ActiveRecord_Relation" objects, and paging data is a Hash type. You should clone my code and test it.
```
[7] pry(#<MetasploitApiApp>)> data.class
=> Mdm::SessionEvent::ActiveRecord_Relation
[8] pry(#<MetasploitApiApp>)> data.merge({ 'pagination_total' => pagination_total })
ArgumentError: Unknown key: "pagination_total". Valid keys are: :includes, :eager_load, :preload, :select, :group, :order, :joins, :where, :having, :bind, :references, :extending, :unscope, :limit, :offset, :lock, :readonly, :from, :reordering, :reverse_order, :distinct, :create_with, :uniq
from /home/kali-team/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/activesupport-4.2.11.1/lib/active_support/core_ext/hash/keys.rb:75:in `block in assert_valid_keys'
[9] pry(#<MetasploitApiApp>)> 

```